lib/shopping_cart.rb

```diff
@@ -0,0 +1,67 @@
+class ShoppingCart
+
+  attr_reader :name, :capacity, :products, :details
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+    @details = Hash.new(0)
```

Comment:
Details should not be stored, youre doing work in find_details that never gets caleld